subsys/ipc/ipc_service/ipc_service.c

```diff
@@ -81,3 +81,119 @@ int ipc_service_send(struct ipc_ept *ept, const void *data, size_t len)
 
 	return backend->send(ept->instance, ept->token, data, len);
 }
+
+int ipc_service_get_tx_buffer(struct ipc_ept *ept, void **data, uint32_t *len, k_timeout_t wait)
+{
+	const struct ipc_service_backend *backend;
+
+	if (!ept || !data || !len) {
+		LOG_ERR("Invalid endpoint, data or len pointer");
+		return -EINVAL;
+	}
+
+	backend = ept->instance->api;
+
+	if (!backend || !backend->get_tx_buffer) {
```

Comment:
when backend has no `get_tx_buffer` pointer shouldn't it return `-ENOTSUP`?

Reply:
If you are implementing a backend and you do not want to implement `get_rx_buffer` for example you can add the hook and return `-ENOTSUP`. The code in `ipc_service.c` is a failsafe to catch the cases when the user didn't implement the hook to avoid an application crash.